src/backend/index.ts

```diff
@@ -2,16 +2,633 @@ import express from 'express';
+                  height: 300,
+                  url: 'https://i.scdn.co/image/ab67616d00001e0209d2915d50e59087f339c17b',
+                  width: 300,
+                },
+                {
+                  height: 64,
+                  url: 'https://i.scdn.co/image/ab67616d0000485109d2915d50e59087f339c17b',
+                  width: 64,
+                },
+              ],
+              name: 'KING',
+              release_date: '2020-08-01',
+              release_date_precision: 'day',
+              total_tracks: 1,
+              type: 'album',
+              uri: 'spotify:album:55F9mwBpB9DRA9g52b8ZGJ',
+            },
+            artists: [
+              {
+                external_urls: {
+                  spotify: 'https://open.spotify.com/artist/1k5LyiTCRzPjORzcgHqJxF',
+                },
+                href: 'https://api.spotify.com/v1/artists/1k5LyiTCRzPjORzcgHqJxF',
+                id: '1k5LyiTCRzPjORzcgHqJxF',
+                name: 'Kanaria',
+                type: 'artist',
+                uri: 'spotify:artist:1k5LyiTCRzPjORzcgHqJxF',
+              },
+            ],
+            disc_number: 1,
+            duration_ms: 134750,
+            episode: false,
+            explicit: false,
+            external_ids: {
+              isrc: 'JPX562000062',
+            },
+            external_urls: {
+              spotify: 'https://open.spotify.com/track/1y70lFnhDN7rPmrl15BtPd',
+            },
+            href: 'https://api.spotify.com/v1/tracks/1y70lFnhDN7rPmrl15BtPd',
+            id: '1y70lFnhDN7rPmrl15BtPd',
+            is_local: false,
+            is_playable: true,
+            linked_from: {
+              external_urls: {
+                spotify: 'https://open.spotify.com/track/5lojBXokpBOfuTDly9Bksx',
+              },
+              href: 'https://api.spotify.com/v1/tracks/5lojBXokpBOfuTDly9Bksx',
+              id: '5lojBXokpBOfuTDly9Bksx',
+              type: 'track',
+              uri: 'spotify:track:5lojBXokpBOfuTDly9Bksx',
+            },
+            name: 'KING',
+            popularity: 45,
+            preview_url:
+              'https://p.scdn.co/mp3-preview/815b3174c42dfc33e079a6b46c56adfe5da04b17?cid=774b29d4f13844c495f206cafdad9c86',
+            track: true,
+            track_number: 1,
+            type: 'track',
+            uri: 'spotify:track:1y70lFnhDN7rPmrl15BtPd',
+          },
+          video_thumbnail: {
+            url: null,
+          },
+        },
+      ],
+      limit: 100,
+      next: null,
+      offset: 0,
+      previous: null,
+      total: 5,
+    },
+    type: 'playlist',
+    uri: 'spotify:playlist:5lclPHYBPxhQnoPUkMjLKZ',
+  } as SpotifyPlaylist;
+  const tracks = data.tracks.items.map((item) => item.track);
+  const albums = tracks.map((track) => track.album);
+
+  const saveAlbumPromises = albums.map(saveAlbum);
+  await Promise.all(saveAlbumPromises);
+  tracks.forEach((track) => {
+    track.albumRef = fireStore.doc(`Albums/${track.album.id}`);
+  });
```

Comment:
```suggestion
  const trackCollections = tracks.map((track) => ({
    ...track
    albumRef: fireStore.doc(`Albums/${track.album.id}`),
  }));
```
既存のリストの破壊的変更はあんまり好かない

Reply:
```
  const trackWithAlbum = tracks.map((track) => {
    track.albumRef = fireStore.doc(`Albums/${track.album.id}`);
    return track;
  });
```
こうしました

Reply:
これ結局破壊してるよねって思った

Reply:
suggestionの書き方を採用しない理由がわからない
L593の末尾にコンマがないのはミスですが

Reply:
採用しました

Reply:
複数行になってくると return が無いと読みづらいと思うんだけどどうでしょう

Reply:
個人的にはまだ2,3行だし、return省略できる言語に比べたら全然マシやろというテンションですがお任せします